akka-actor-typed/src/main/scala/akka/actor/typed/internal/adapter/ActorAdapter.scala

```diff
@@ -222,7 +222,9 @@ import akka.annotation.InternalApi
       case null                   => // skip PostStop
       case _: DeferredBehavior[_] =>
       // Do not undefer a DeferredBehavior as that may cause creation side-effects, which we do not want on termination.
-      case b => Behavior.interpretSignal(b, ctx, PostStop)
+      case b =>
+        if (_ctx eq null) initializeContext()
```

Comment:
could you repeat why it's important to not create the context in all cases from `preStart`?

Reply:
note that `GuardianActorAdapter` is also messing with `initializeContext()`, I think it needs a holistic cleanup

Reply:
and when cleaning up that also take a look at `protected var behavior` and see if that can be made `private`, now that we have `def currentBehavior`

Reply:
Yeah, I've been on the fence about doing a bigger rework of it, I don't like it at all how hard it is to follow, but I also remember that it was really tricky to get it behave like we want it to when it was created.

I wonder if we could _just_ make it a stashing behavior instead and not have a special adapter at all... (I have a feeling I tried that back then though)

Reply:
Re the avoiding of creating the context I have no idea, let's create it all the time.